File: mysql_examples/recipes/api/05_null_in_result/null-in-result.php
<?php
# null-in-result.php - print undef values as "NULL"

require_once "Cookbook.php";

$conn =& Cookbook::connect ();
if (PEAR::isError ($conn))
  die ("Cannot connect to server\n");

#@ _FETCHLOOP_1_
$result =& $conn->query ("SELECT name, birth, foods FROM profile");
if (PEAR::isError ($result))
  die ("Oops, the statement failed");
while ($row =& $result->fetchRow ())
{
  foreach ($row as $key => $value)
  {
    if ($row[$key] === NULL)
      $row[$key] = "NULL";
  }
  print ("name: $row[0], birth: $row[1], foods: $row[2]\n");
}
$result->free ();
#@ _FETCHLOOP_1_

$conn->disconnect ();
?>

File: mysql_examples/recipes/api/05_null_in_result/null-in-result.py
#!/usr/bin/python

import Cookbook

conn = Cookbook.connect ()

try:
#@ _FETCHLOOP_
  cursor = conn.cursor ()
  cursor.execute ("SELECT name, birth, foods FROM profile")
  for row in cursor.fetchall ():
    row = list (row)  # convert nonmutable tuple to mutable list
    for i in range (0, len (row)):
      if row[i] == None:  # is the column value NULL?
        row[i] = "NULL"
    print "name: %s, birth: %s, foods: %s" % (row[0], row[1], row[2])
  cursor.close ()
#@ _FETCHLOOP_
except:
  print "Oops, the statement failed"

conn.close ()
